### PR TITLE
Add Iterative ML tools feature collection

### PR DESCRIPTION
**Pull request by shcheklein**
Adds [Iterative](https://iterative.ai) feature collection per [this document](https://code.visualstudio.com/blogs/2022/09/15/dev-container-features#_distribution).

Collection includes the [Data Version Control](https://github.com/iterative/dvc) feature as a first step, a popular tool in the ML/MLOps domain.

It simplifies devcontainer setup for Codespaces significantly, thanks for the great new functionality Here is [an example](https://github.com/shcheklein/example-get-started/blob/main/.devcontainer.json):

```json
{
  "name": "example-get-started",
  "image": "mcr.microsoft.com/devcontainers/python:3",
  "extensions": ["Iterative.dvc"],
  "features": {
      "ghcr.io/iterative/features/dvc:1": {}
  },
  "postCreateCommand": "pip3 install --user -r src/requirements.txt"
}
```

It's now a single file instead of a directory with Dockerfile.

### CODE REVIEW

File: _data/collection-index.yml [hunk 1]
```diff
@@ -3,3 +3,9 @@
   contact: https://github.com/devcontainers/features/issues
   repository: https://github.com/devcontainers/features
   ociReference: ghcr.io/devcontainers/features
+- name: Iterative Tools for Machine Learning Features 
+  maintainer: Iterative, Inc
+  contact: https://github.com/iterative/features/issues
+  repository: https://github.com/iterative/features
+  ociReference: ghcr.io/iterative/features
+
```
Comment (samruddhikhandale) [inline _data/collection-index.yml]: nit: Can we remove this line?

**Review by samruddhikhandale — COMMENTED** [review]
Thank you for the opening the PR  🎉  The `dvc` Feature looks super cool.
Left one comment. 😄

**Review by joshspicer — APPROVED** [review]
Awesome! 🚀

File: _data/collection-index.yml (hunk 1 above)
Comment (shcheklein) [reply _data/collection-index.yml]: done, sorry about that, should be fixed now

**Comment by joshspicer** [comment]
FYI, if you'd like, you can simplify that `devcontainer.json` even further by including the VS Code extension in your Feature.

As an example: https://github.com/devcontainers/features/blob/main/src/aws-cli/devcontainer-feature.json#L17-L22

Then you could remove that extension from your example `devcontainer.json` above 🚀

**Comment by shcheklein** [comment]
@joshspicer that's exactly what I was about to ask / feature request. It's good to know that it's possible. But after that decided to keep it split for now (it's not that higher in complexity but gives a bit more flexibility / less magic). We'll see how it goes and if we want to bundle them together.

**Comment by joshspicer** [comment]
sounds good! @shcheklein 

All of these properties are supported (with discussions in the devcontainers/spec repo for adding more): https://containers.dev/implementors/features/#devcontainer-feature-json-properties

**Pull request merged**